File: 1101-1199/1179/1179.py
import sys

a = [False]*37
b = [0]*37

for ch in sys.stdin.read():
	if '0' <= ch <= '9':
		j = ord(ch) - ord('0')
	elif 'A' <= ch <= 'Z':
		j = ord(ch) - ord('A') + 10
	else:
		j = sys.maxint
	for i in range(2, 37):
		if j < i:
			if not a[i]:
				b[i] += 1
				a[i] = True
		else:
			if a[i]:
				a[i] = False

maxbase, maxnum = 0, 0
for i in range(2, 37):
	if b[i] > maxnum:
		maxbase = i
		maxnum = b[i]

print maxbase, maxnum

File: 1101-1199/1179/1179.c
#include <stdio.h>

int main()
{
	int a[37];
	int b[37];
	int c, i;
	for (i=2; i<=36; i++) 
	{
		a[i] = 0;
		b[i] = 0;
	}
	while ((c = getchar()) != EOF)
	{
		if ((int)'0' <= c && c <= (int)'9')
			c -= (int)'0';
		else if ((int)'A' <= c && c <= (int)'Z')
			c -= (int)'A' - 10;
		else
			c = 9999;
		for (i=2; i<=36; i++)
		{
			if (c < i)
			{
				if (a[i] == 0)
				{
					b[i] ++;
					a[i] = 1;
				}
			}
			else
			{
				if (a[i] == 1)
					a[i] = 0;
			}
		}
	}
	int maxbase = 0, maxnum = 0;
	for (i=2; i<=36; i++)
	{
		if (b[i] > maxnum)
		{
			maxbase = i;
			maxnum = b[i];
		}
	}
	printf("%d %d", maxbase, maxnum);
}
